### PR TITLE
the result of the to function may be empty, this is not what we want

### PR DESCRIPTION
**Pull request by xhuangdx**
### Description
The result of the to() function may be empty, which can have serious consequences, and should be avoided at this time.

### Type of change
- [x] Bug Fix

### CODE REVIEW

**Comment by mzhangdx** [comment]
> How could transaction.To being nil? And please add related test cases to improve code coverage.

The `to` will be nil when creating smart contract: `core/types/transaction.go:53`

**Comment by jwangdx** [comment]
> > How could transaction.To being nil? And please add related test cases to improve code coverage.
> 
> The `to` will be nil when creating smart contract: `core/types/transaction.go:53`

Got it. Thanks

**Comment by xhuangdx** [comment]
> Please add test cases for parts you modified. Other than that, everything looks fine.

Thank you for your reply, I will add test case later.

File: storage/storagehost/hostheightchange_test.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+		GasLimit:   uint64(5000),
+		GasUsed:    uint64(300),
+		Time:       big.NewInt(1550103878),
+		Extra:      []byte{},
+		MixDigest:  crypto.Keccak256Hash(nil),
+		Nonce:      types.EncodeNonce(uint64(1)),
+	}
+}
+
+func (m *mokeHostBackend) GetBlockByHash(blockHash common.Hash) (*types.Block, error) {
```
Comment (jwangdx) [inline storage/storagehost/hostheightchange_test.go]: Spelling error: moke should be mock

Comment (xhuangdx) [reply storage/storagehost/hostheightchange_test.go]: Resolved, thank you for your reminder

**Pull request merged**